src/utils/auth.ts

```diff
@@ -0,0 +1,22 @@
+import express from 'express';
+const jwt = require('jsonwebtoken');
+
+// TODO: Test out this middle where once login endpoint is done and returns a JSON web token
+const authenticateJWT = (req: any, res: any, next: () => void) => {
+    const authHeader = req.headers.authorization;
+
+    if (authHeader) {
+        const token = authHeader.split(' ')[1];
+        try {
+            jwt.verify(token, process.env.ACCESS_TOKEN_SECRET)
```

Comment:
If the token is valid, you need to add `req.user` and load the user for that token
```suggestion
            const user = jwt.verify(token, process.env.ACCESS_TOKEN_SECRET)
            req.user = await getUser(user.id);
```

Reply:
Right you may need to do `await jwt.verify`

Reply:
Is `getUser(user.id)` just a placeholder function? I just added `req.user = user.id` on that line because I awaited the assignment of `user` from `jwt.verify`